autokeras/auto/auto_model.py

```diff
@@ -18,59 +22,43 @@ class AutoModel(object):
         tuner: An instance of Tuner.
     """
 
-    def __init__(self, inputs, outputs, tuner=None):
+    def __init__(self, **kwargs):
         """
         """
-        self.inputs = format_inputs(inputs)
-        self.outputs = format_inputs(outputs)
-        self.hypermodel = None
-        self.tuner = tuner
-        self.optimizer = None
-        self.metrics = None
-        self.loss = None
-
-    def hyperparameters(self):
-        pass
-
-    def compile(self,
-                optimizer=None,
-                metrics=None,
-                loss=None):
-        self.optimizer = optimizer
-        self.metrics = metrics
-        self.loss = loss
+        super().__init__(**kwargs)
+        self.inputs = []
+        self.outputs = []
+        self.tuner = tuner.SequentialRandomSearch(self, objective=self.get_metrics())
+
+    def build(self, hp):
+        raise NotImplementedError
 
     def fit(self,
             x=None,
             y=None,
             validation_data=None,
-            tuner=None,
-            trails=None,
+            num_trials=None,
             **kwargs):
         # Initialize HyperGraph model
-        x = format_inputs(x, 'train_x')
-        y = format_inputs(y, 'train_y')
+        x = layer_utils.format_inputs(x, 'train_x')
+        y = layer_utils.format_inputs(y, 'train_y')
+
+        # TODO: Set the shapes only if they are not provided by the user when initiating the HyperHead or Block.
         for x_input, input_node in zip(x, self.inputs):
             input_node.shape = x_input.shape[1:]
         for y_input, output_node in zip(y, self.outputs):
             if len(y_input.shape) == 1:
                 y_input = np.reshape(y_input, y_input.shape + (1,))
             output_node.shape = y_input.shape[1:]
-        self.hypermodel = HyperGraph(self.inputs, self.outputs)
-
-        # Initialize Tuner
-        if tuner is not None:
-            self.tuner = tuner
-        else:
-            self.tuner = SequentialRandomSearch(self.hypermodel,
-                                                objective=self.metrics)
+            output_node.in_hypermodels[0].output_shape = output_node.shape
+
         # Prepare the dataset
         if validation_data is None:
-            (x, y), (x_val, y_val) = split_train_to_valid(x, y)
+            (x, y), (x_val, y_val) = layer_utils.split_train_to_valid(x, y)
             validation_data = x_val, y_val
 
         # TODO: allow early stop if epochs is not specified.
-        self.tuner.search(trails,
+        self.tuner.search(num_trials=num_trials,
```

Comment:
I would recommend just naming this `trials` (like we have `epochs` or `steps`, for instance). The `num_` prefix is only useful if there is some potential confusion about the meaning of the name. For instance, if you have an `int` argument `layers`, that is confusing (it could be a list of layer instances?), so better to use `num_layers` in that case.

But for `trials` I think it is unambiguous that it is a number.